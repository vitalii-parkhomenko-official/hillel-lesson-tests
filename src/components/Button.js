function Button(props) {
  const { type = "button" } = props;
  const variant = props.variant || "primary";
  const className = props.className ? `btn btn-${variant} ${props.className}` : `btn btn-${variant}`;

  return (
    <button
      data-testid={props["data-testid"] || "button"}
      type={type}
      className={className}
      onClick={props.onClick}
    >
      {props.children}
    </button>
  );
}

export default Button;
