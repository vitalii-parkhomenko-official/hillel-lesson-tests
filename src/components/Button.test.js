import { render, fireEvent } from "@testing-library/react";
import Button from "./Button";

describe("Button", () => {
  test("render default", () => {
    const { getByTestId } = render(
      <Button>
        Test
      </Button>
    );

    const btn = getByTestId("button");

    expect(btn.tagName).toEqual("BUTTON");
    expect(btn.type).toEqual("button");
    expect(btn.className).toMatch("btn btn-primary");
    expect(btn.textContent).toEqual("Test");
  });

  test("render snapshot", () => {
    const { getByTestId } = render(
      <Button>
        Test
      </Button>
    );

    const btn = getByTestId("button");

    expect(btn).toMatchSnapshot();
  });

  test("render secondary", () => {
    const { getByTestId } = render(
      <Button variant="secondary">
        Test
      </Button>
    );

    const btn = getByTestId("button");

    expect(btn.tagName).toEqual("BUTTON");
    expect(btn.type).toEqual("button");
    expect(btn.className).toEqual("btn btn-secondary");
  });

  test("render with custom class", () => {
    const { getByTestId } = render(
      <Button className="ms-3">
        Test
      </Button>
    );

    const btn = getByTestId("button");

    expect(btn.tagName).toEqual("BUTTON");
    expect(btn.type).toEqual("button");
    expect(btn.className).toEqual("btn btn-primary ms-3");
  });

  test("call on click", () => {
    const onClick = jest.fn();
    const { getByTestId } = render(
      <Button onClick={onClick}>
        Test
      </Button>
    );

    const btn = getByTestId("button");

    fireEvent.click(btn);

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
