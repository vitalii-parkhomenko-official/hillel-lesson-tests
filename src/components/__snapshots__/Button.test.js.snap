// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render snapshot 1`] = `
<button
  class="btn btn-primary"
  data-testid="button"
  type="button"
>
  Test
</button>
`;
